fdm-app/app/integrations/nmi.ts

```diff
@@ -27,7 +28,13 @@ export async function getSoilParameterEstimates(
         throw new Error("Please provide a NMI API key")
     }
 
-    const fieldCentroid = centroid(field.geometry)
+    let geometry: Geometry
+    if ("geometry" in field) {
+        geometry = field.geometry
+    } else {
+        geometry = field
+    }
```

Comment:
you are certain that field is a spatial object, when the case occurs that geometry is not in field?